### ISSUE TITLE
ABC Portal doesn't allow people records w/ different countries (On PROD) (NN)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#10505]
### Portal won't allow customers to register with an out-of-country address. This complicates things & causes customers to contact staff to be fixed on the backend. 
- INC0281660 Created
- Pending Portal Subcommittee approved in March 2023

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT  analyze and let me know if it has to go to ctm/vendor

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader This needs to go ctm/vendor. Changes/additions need to be done in portal registration step 2: user information page to allow user to enter out-of-country addresses.

**Comment by Lpriya-LIT** [comment]
![image.png](https://images.zenhubusercontent.com/611eac1954217f265acc0499/cc5ea7f3-c8f3-4a36-a2d0-77b2e4701f60)

**Comment by Nadin-Nader** [comment]
@TracyLinder  @Lpriya-LIT i have sent this to CTM to speak to the vendor about under **SCTASK0258822**

**Comment by Nadin-Nader** [comment]
@TracyLinder @Lpriya-LIT I spoke to Richard R. from the Portal committee and he said he would add this to the next Portal committee meeting (11/14/22) for Discussion and Approval.

**Comment by Nadin-Nader** [comment]
Swapna reached out to me and said that she was working on this and needed me to UAT DEV to see if this is what the user wants.

**Comment by Nadin-Nader** [comment]
UATd and found that when i tried to register the account there were only two options for Country, United States and Europe. If Europe is selected , at first, it greys out the State drop down. Then if i have to go back to correct something (i intentionally didnt put my last name), i was able to look through the State list. It did not affect selecting the "Europe" Option. I also added a Paris address.
I noticed that it still asked for a zipcode. I will need to ask about this since other countries call it a "Postal Code".
I also noticed that the phone format is still the same (Area code-then the 7 number). Some countries have a different format and country code, i will need to ask Paloma if leaving it as is will affect it.

**Comment by Nadin-Nader** [comment]
I had Paloma look over the DEV version of what was done and she came back with the following results on a screenshot:

![MicrosoftTeams-image (33).png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/176def41-a464-44dd-a70e-9b28ce3bc186)

I will relay the message to Swapna

**Comment by Nadin-Nader** [comment]
Per Swapna, we will need to wait til the next committee meeting to discuss any further changes

**Comment by TracyLinder** [comment]
### Portal Conversation

**Fix:**
Conditional validation on country
- US. then validate zip code is 5 digits. 
- Europe then allow up to 5 digits.

**Comment by TracyLinder** [comment]
I need to submit a new ticket for this.

**Comment by TracyLinder** [comment]
Submitted SN ticket, updated description, and moved to On Deck.

**Comment by Nadin-Nader** [comment]
From Swapna's email this is ready to test on TEST Portal. I was not able to test as the login and register feature is giving me issues. I will ask Kyle via TEAMS to UAT

**Comment by Nadin-Nader** [comment]
Sent Message to Kyle and ROW Team to please UAT and let us know if this is good to send to PROD or not. Waiting on Response

**Comment by TracyLinder** [comment]
**Update:**
CTM says there is a bug with this in the back office that they're working on.

**Comment by Nadin-Nader** [comment]
Ready for UAT on TEST

**Comment by Nadin-Nader** [comment]
Relayed the following message from Kyle at ROW to Swapna and Matthew. Waiting on their response and/or action:

> [Yesterday 11:43 AM] Kampe, Kyle
>     I think my only major concern is that the list of countries doesn't appear to be comprehensive. I only counted 100 options, even though there's around 195 countries. Glaring omissions are Mexico and the United Kingdom.
>

**Comment by TracyLinder** [comment]
Per CTM, the fixes have been made and this is ready to test again.

**Comment by Nadin-Nader** [comment]
This is being deployed to PROD. Waiting on Deployment confirmation

**Comment by Nadin-Nader** [comment]
Message sent to Kyle that this is on PROD and ready for UAT

**Comment by Nadin-Nader** [comment]
Closing as Kyle has approved this on PROD
